src/ResourceManager/AnalysisServices/Commands.AnalysisServices.ServiceManagement/Models/AsAzureEnvironment.cs

```diff
@@ -0,0 +1,39 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.Commands.AnalysisServices.ServiceManagement
+{
+    [Serializable]
+    public class AsAzureEnvironment
+    {
+        public AsAzureEnvironment(string Name)
+        {
+            this.Name = Name;
+            this.Endpoints = new Dictionary<AsRolloutEndpoints, string>();
+        }
+
+        public string Name { get; set; }
+
+        public Dictionary<AsRolloutEndpoints, string> Endpoints { get; set; }
```

Comment:
Dictionary is not particularly serializable in powershell, consider using a Hashtable instead.